src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs

```diff
@@ -29,13 +29,19 @@ internal unsafe static class UnicodeHelpers
 
         /// <summary>
         /// Helper method which creates a bitmap of all characters which are
-        /// defined per version 7.0.0 of the Unicode specification.
+        /// defined per the Unicode specification.
         /// </summary>
         [MethodImpl(MethodImplOptions.NoInlining)]
         private static uint[] CreateDefinedCharacterBitmap()
         {
             // The stream should be exactly 8KB in size.
-            var stream = typeof(UnicodeHelpers).GetTypeInfo().Assembly.GetManifestResourceStream("System.Text.Encodings.Web.Resources.unicode-8.0.0-defined-characters.bin");
+            var stream = typeof(UnicodeRange).GetTypeInfo().Assembly.GetManifestResourceStream("System.Text.Encodings.Web.Resources.unicode8definedcharacters.bin");
+
+            if (stream == null)
+            {
+                throw new BadImageFormatException();
```

Comment:
I actually think there is a difference between these. The resource might be missing for configuration reasons. e.g. satellite assembly missing. The other checks can only fail if we produced the stream incorrectly or it got corrupted on disk.

Also, I was trying to minimize the churn in this change. I don't think this is a good time to change the rest of the routine.